lib/osf-components/addon/components/draft-registration-card/template.hbs

```diff
@@ -42,55 +46,62 @@
                 </p>
             </small>
             <div class='row'>
-                <div class='col-md-10'>
-                    <OsfLink
-                        data-analytics-name='Edit'
-                        class='btn btn-default'
-                        @route='registries.drafts.draft'
-                        @models={{array @draftRegistration.id}}
-                    >
-                        {{fa-icon 'pencil-alt'}} {{t 'general.edit'}}
-                    </OsfLink>
-                    <OsfButton
-                        data-analytics-name='Delete'
-                        @type='default'
-                        @onClick={{action this.delete}}
-                        disabled={{not this.draftRegistration}}
-                    >
-                        {{fa-icon 'times'}} {{t 'general.delete'}}
-                    </OsfButton>
-                    <BsModal
-                        @open={{this.deleteModalOpen}}
-                        @onHidden={{action this.cancelDelete}}
-                        class='delete_draft_registration'
-                        as |modal|
-                    >
-                        <modal.header>
-                            <h4>{{t 'general.please_confirm'}}</h4>
-                        </modal.header>
-                        <modal.body>
-                            {{t 'osf-components.draft-registration-card.delete_draft_confirm'}}
-                        </modal.body>
-                        <modal.footer data-analytics-scope='Draft registrations delete modal'>
+                {{#unless this.draftRegistration.currentUserIsReadOnly}}
+                    <div class='col-md-10'>
+                        <OsfLink
+                            data-test-draft-card-edit
+                            data-analytics-name='Edit'
+                            class='btn btn-default'
+                            @route='registries.drafts.draft'
+                            @models={{array @draftRegistration.id}}
+                        >
+                            {{fa-icon 'pencil-alt'}} {{t 'general.edit'}}
+                        </OsfLink>
+                        {{#if this.draftRegistration.currentUserIsAdmin}}
                             <OsfButton
-                                data-analytics-name='Cancel delete'
-                                @onClick={{action modal.close}}
+                                data-test-draft-card-delete
+                                data-analytics-name='Delete'
                                 @type='default'
+                                @onClick={{action this.delete}}
+                                disabled={{not this.draftRegistration}}
```

Comment:
Is this needed? If there is not a `draftRegistration`, then the whole card shouldn't be there.